### PR TITLE
chore: restructured deep.spec.ts

### PR DESCRIPTION
**Pull request by splincode**
## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [x] Refactoring
- [ ] Build or CI related changes
- [ ] Tests related changes
- [ ] Documentation content changes

### CODE REVIEW

File: projects/demo-playwright/tests/deep/deep.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
+import {Locator, Page, test} from '@playwright/test';
+
+export class DeepSpecHelpers {
```
Comment (nsbarsukov) [inline projects/demo-playwright/tests/deep/deep.ts:3]: Should we transform it to `TuiDocumentationAPIPagePO`?

```ts
const apiPage = new TuiDocumentationAPIPagePO(page);
```

File: projects/demo-playwright/tests/deep/deep.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+    static getDemoContent(page: Page): Locator {
+        return page.locator('#demo-content');
+    }
+
+    static async waitSomeChanges(page: Page): Promise<void> {
+        await page.evaluate(
+            async () =>
+                new Promise(resolve => {
+                    setTimeout(resolve, 200);
```
Comment (nsbarsukov) [inline projects/demo-playwright/tests/deep/deep.ts:70]: `await page.waitForTimeout(200)` ?

File: projects/demo-playwright/tests/deep/deep.ts (hunk 1 above)
Comment (splincode) [reply projects/demo-playwright/tests/deep/deep.ts]: fixed

File: projects/demo-playwright/tests/deep/deep.ts (hunk 2 above)
Comment (splincode) [reply projects/demo-playwright/tests/deep/deep.ts]: fixed

File: projects/demo-playwright/utils/goto.ts [hunk 3]
```diff
@@ -40,9 +40,6 @@ export async function tuiGoto(
 
     const response = await page.goto(url, playwrightGotoOptions);
 
-    await page.waitForLoadState('domcontentloaded');
-    await page.waitForLoadState('load');
```
Comment (nsbarsukov) [inline projects/demo-playwright/utils/goto.ts:44]: Why ?

**Review by nsbarsukov — APPROVED** [review]

File: projects/demo-playwright/utils/goto.ts (hunk 3 above)
Comment (splincode) [reply projects/demo-playwright/utils/goto.ts]: revert

**Pull request merged**